specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2018-02-01-preview/containerregistry_build.json

```diff
@@ -0,0 +1,2242 @@
+          "type": "string"
+        },
+        "buildArguments": {
+          "description": "The custom arguments for building this build step.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BuildArgument"
+          }
+        },
+        "baseImageDependencies": {
+          "description": "List of base image dependencies for a step.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BaseImageDependency"
+          },
+          "readOnly": true
+        },
+        "baseImageTrigger": {
+          "description": "The type of the auto trigger for base image dependency updates.",
+          "enum": [
+            "All",
+            "Runtime",
+            "None"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "BaseImageTriggerType",
+            "modelAsString": true
+          }
+        }
+      },
+      "x-ms-discriminator-value": "Docker"
+    },
+    "BaseImageDependency": {
+      "description": "Properties that describe a base image dependency.",
+      "required": [
+        "repositoryName",
+        "digest"
+      ],
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "The type of the base image dependency.",
+          "enum": [
+            "BuildTime",
+            "RunTime"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "BaseImageDependencyType",
+            "modelAsString": true
+          }
+        },
+        "isAutoTriggerEnabled": {
+          "description": "The value of this flag indicates if build has to be auto triggered when this image is updated.",
+          "default": false,
+          "type": "boolean"
+        },
+        "repositoryName": {
+          "description": "Name of the repository.",
+          "type": "string"
+        },
+        "tag": {
+          "description": "The tag name.",
+          "type": "string"
+        },
+        "digest": {
+          "description": "The image digest. Example: SHA256 based digest.",
+          "type": "string"
+        }
+      }
+    },
+    "DockerBuildStepUpdateParameters": {
+      "description": "The properties for updating a docker build step.",
+      "required": [
+        "type"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BuildStepPropertiesUpdateParameters"
+        }
+      ],
+      "properties": {
+        "branch": {
+          "description": "The repository branch name.",
+          "type": "string"
+        },
+        "imageName": {
+          "description": "TODO (bindu): Backward-compatibility, bring back ImageName to support old CLI build extension, need to remove it.",
+          "type": "string"
+        },
+        "imageNames": {
+          "description": "The fully qualified image names including the repository and tag.",

```

Comment:
Clean up this description? Why is it showing in swagger? /cc @northtyphoon

Reply:
AS per talking to @ankurkhemani  this should exist for backward compatibility of old CLI extension.

Reply:
Talked to Sajay and I will remove the obsolete properties